.github/workflows/appsignals-python-e2e-test.yml

```diff
@@ -0,0 +1,321 @@
+# This is a reusable workflow for running the Python E2E test for App Signals.
+# It is meant to be called from another workflow.
+# This E2E test is responsible for validating setting up a sample application on an EKS cluster and enabling
+# App Signals using the staging image of the CloudWatch Agent Operator. It validates the generated telemetry
+# including logs, metrics, and traces, then cleans up the cluster. The testing resources can be found in the
+# ADOT python test framework repo: https://github.com/aws-observability/aws-application-signals-test-framework/tree/main
+# Read more about reusable workflows: https://docs.github.com/en/actions/using-workflows/reusing-workflows#overview
+name: App Signals Enablement Python E2E Testing
+on:
+  workflow_call:
+    inputs:
+      # Ensure two tests do not run on the same cluster at the same time through GitHub Action concurrency
+      test-cluster-name:
+        required: true
+        type: string
+
+permissions:
+  id-token: write
+  contents: read
+
+env:
+  AWS_DEFAULT_REGION: us-east-1
+  TEST_ACCOUNT: ${{ secrets.APP_SIGNALS_E2E_TEST_ACCOUNT_ID }}
+  SAMPLE_APP_NAMESPACE: python-sample-app-namespace
+  APP_SIGNALS_PYTHON_E2E_FE_SA_IMG: appsignals-python-django-main-service
+  APP_SIGNALS_PYTHON_E2E_RE_SA_IMG: appsignals-python-django-remote-service
+  METRIC_NAMESPACE: AppSignals
+  LOG_GROUP: /aws/appsignals/eks
+#  ECR_OPERATOR_STAGING_IMAGE: ${{ secrets.ECR_OPERATOR_STAGING_IMAGE }}
+
+
+jobs:
+  appsignals-e2e-test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          # TODO: This step is used for custom pre-release instrumentation
+          # Uisng SHA as of March 4. It is a temporary measure until the cw add-on is released with python support
+          ref: abf75babe672412cb63c56cbcf1c5ce2d8c97a1c
```

Comment:
Yes, the difference it the helm change. I tested helm using main branch, and the CW manager log shows:
```
kubectl logs -f amazon-cloudwatch-observability-controller-manager-867b6b8fztzf -n amazon-cloudwatch
{"level":"info","ts":"2024-03-15T21:13:21Z","msg":"Starting the Amazon CloudWatch Agent Operator","amazon-cloudwatch-agent-operator":"","cloudwatch-agent":"public.ecr.aws/cloudwatch-agent/cloudwatch-agent:0.0.0","auto-instrumentation-java":"%!s(<nil>)/%!s(<nil>):%!s(<nil>)","auto-instrumentation-python":"%!s(<nil>)/%!s(<nil>):%!s(<nil>)","build-date":"","go-version":"go1.20.14","go-arch":"amd64","go-os":"linux"}
```
Then, when I restart my sample app pods, I get:
```
django-frontend-service-58c68f4998-clbvp     0/1     Init:InvalidImageName   0          79s
image-backend-6bd56fd569-2lmdg               0/1     Init:InvalidImageName   0          79s
vehicle-inventory-backend-59db98c8d8-dlc92   0/1     Init:InvalidImageName   0          79s
```
Seems both adot Java and python images values are  nil. Does this mean the operator doesn’t find the correct image, and failed with injection?

Also, I think I need to use staging image, since the test will be added into staging artifacts build.